### PR TITLE
Fixes to allow ARM to cross compile

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,10 +7,11 @@ on:
       - '!master'
 
 env:
-  BUILDER_VERSION: v0.4.5
+  BUILDER_VERSION: cpp-fixes
```
Comment (graebm) [inline .github/workflows/ci.yml]: change before merge?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,7 +28,7 @@ jobs:
         echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u awslabs --password-stdin
         export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-crt-builder/aws-crt-${{ matrix.image }}:${{ env.BUILDER_VERSION }}
         docker pull $DOCKER_IMAGE
-        docker run --env GITHUB_REF $DOCKER_IMAGE build -p ${{ env.PACKAGE_NAME }}
+        docker run --env GITHUB_REF $DOCKER_IMAGE build -p ${{ env.PACKAGE_NAME }} --dump-config
```
Comment (graebm) [inline .github/workflows/ci.yml]: is `--dump-config` a debugging thing, do you want to keep this here when you merge?

File: .gitmodules [hunk 3]
```diff
@@ -21,4 +21,4 @@
 	url = https://github.com/awslabs/aws-c-mqtt.git
 [submodule "aws-common-runtime/s2n"]
 	path = aws-common-runtime/s2n
-	url = https://github.com/awslabs/s2n.git
+	url = https://github.com/justinboswell/s2n.git
```
Comment (graebm) [inline .gitmodules]: change before merge?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (justinboswell) [reply .github/workflows/ci.yml]: Yes.

**Comment by justinboswell** [comment]
> there's a lot of stuff in here that I suspect we don't want merged to master?
> 
> such as the s2n fork in your personal github account?

Oh crap, I need to revert that. Will fix pronto.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (justinboswell) [reply .github/workflows/ci.yml]: I can remove it. It doesn't hurt anything either way.

**Pull request merged**